server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1250,6 +1250,39 @@ private static boolean isFiltered(Property[] properties) {
         return properties != null && Arrays.asList(properties).contains(Property.Filtered);
     }
 
+    /**
+     * A writeable validator able to check the value of string type custom setting by using regular expression
+     */
+    public static class RegexValidator implements Writeable, Validator<String> {
+        private Pattern pattern;
+
+        public RegexValidator(String regex) {
+            this.pattern = Pattern.compile(regex);
+        }
+
+        public RegexValidator(StreamInput in) throws IOException {
+            this.pattern = Pattern.compile(in.readString());
+        }
+
+        Pattern getPattern() {
+            return pattern;
+        }
+
+        @Override
+        public void validate(String value) {
+            if (!pattern.matcher(value).matches()) {
+                throw new IllegalArgumentException(
+                    "Setting must be valid [" + value + "],Custom setting matches regex [" + pattern.pattern() + "]"
```

Comment:
This message doesn't match the failure case. I think I suggested this in an issue or PR comment somewhere else when I thought you were failing if it matched the regex.  With the `!` you are failing if it does NOT match.  That's fine but this error message should be `"Setting [" + value + "] does not match regex [" + pattern.pattern() + "]"`.